src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,241 @@
+  result: Result;
+}
+
+class TimeOfNeed extends Analyzer {
+  spawns: tonEvent[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.TIME_OF_NEED_TALENT);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.summon.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.TIME_OF_NEED_SUMMON),
+      this.onSpawn,
+    );
+  }
+
+  onSpawn(event: SummonEvent) {
+    const HealEvents = getTimeOfNeedHealing(event);
+
+    const currentEvent: tonEvent = {
+      summon: event,
+      verdantEmbrace: null,
+      livingFlames: [],
+      livingFlameTotalHeal: 0,
+      livingFlameTotalOverheal: 0,
+      target: 0,
+      result: Result.Not,
+    };
+
+    HealEvents.forEach((heal) => {
+      if (heal.ability.guid === SPELLS.VERDANT_EMBRACE_HEAL.id) {
+        currentEvent.verdantEmbrace = heal;
+        currentEvent.target = heal.targetID;
+      } else if (heal.ability.guid === SPELLS.TIME_OF_NEED_LIVING_FLAME.id) {
+        currentEvent.livingFlames.push(heal);
+        currentEvent.livingFlameTotalHeal += heal.amount + (heal.absorbed || 0);
+        currentEvent.livingFlameTotalOverheal += heal.overheal ?? 0;
+      }
+    });
+
+    this.spawns.push(currentEvent);
+  }
+
+  parseDamageTaken(tonSummon: SummonEvent, eventList: WCLEventsResponse) {
+    //Iterate through all Time of Need spawns to find the one that matches
+    this.spawns.forEach((spawn, index) => {
+      if (spawn.summon.timestamp === tonSummon.timestamp) {
+        //Check that ToN actually did any healing
+        if (spawn.verdantEmbrace !== null || spawn.livingFlames.length > 0) {
+          //If there is no damage events, then Time of Need didn't save the player from anything. Otherwise, parse them
+          if (eventList.events.length !== 0) {
+            //Iterate all the damage taken events during the time window
+            eventList.events.forEach((hit) => {
+              hit = hit as DamageEvent;
+              //Check if the hit landed on the same player that the Time of Need targeted
+              if (hit.targetID === spawn.target) {
+                //If the hitpoints after the hit are equal or less than 0, the player died anyway
+                if (hit.hitPoints !== undefined && hit.hitPoints === 0) {
+                  this.spawns[index].result = Result.Dead;
+                } else {
+                  //Initialize the healing from Time of Need with the VE
+                  let tonHealingAtThisPoint = spawn.verdantEmbrace?.amount || 0;
+                  //If there is no living flames casted, or this hit was before the first living flame landed
+                  if (
+                    spawn.livingFlames.length === 0 ||
+                    hit.timestamp < spawn.livingFlames[0].timestamp
+                  ) {
+                    //Check if the player would've died from this hit without the Verdant Embrace healing
+                    if (
```

Comment:
rather than nesting this 7 `if`s deep, use early returns on the negated conditions to make things easier to follow. your logic is basically:

1. check that there is healing done, if not return `Bug`
2. check that there is damage taken, if not return `Not` (i don't think you currently do this?)
3. go through the damage taken events and see if the healing from ToN prevented a death, either from an individual heal (`Proc`) or from the total healing (`Long`).
    - if they died despite the healing, early return `Dead`
    - if none of the above, `Not`.

that doesn't need to be nested so deeply.

i don't know enough about the ToN talent to know if it makes sense to consider some large amount of healing that didn't directly prevent a death as relevant, i'll leave that up to you + the healer folks.